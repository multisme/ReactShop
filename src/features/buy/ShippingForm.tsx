import React from "react";
import { Link, useHistory } from "react-router-dom";

import { useFormFields, checkValidFields } from "utils/utils";

const ShippingForm = () => {
  const [fields, handleFieldChange] = useFormFields({
    name: {},
    email: {},
    address: {},
    city: {},
    postalCode: {},
  });

  const history = useHistory();

  const handleClick = (e: any) => {
    history.push("/pay", [fields]);
    e.preventDefault();
  };

  return (
    <div id="ShipForm" className={"rectangle"}>
      <form onSubmit={handleClick}>
        <label>
          Name:{" "}
          <input type="text" onChange={handleFieldChange} id="name" required />{" "}
        </label>
        <label>
          Email:{" "}
          <input
            type="email"
            onChange={handleFieldChange}
            id="email"
            required
          />{" "}
        </label>
        <label>
          Shipping Address:{" "}
          <input
            type="text"
            onChange={handleFieldChange}
            id="address"
            required
          />{" "}
        </label>
        <label>
          City:{" "}
          <input type="text" onChange={handleFieldChange} id="city" required />{" "}
        </label>
        <label>
          Postal Code:{" "}
          <input
            type="text"
            onChange={handleFieldChange}
            id="postalCode"
            required
          />{" "}
        </label>
        <fieldset className="button">
          <button type="submit" onSubmit={handleClick}>
            SHIP
          </button>
          <Link to="/home">
            <button>CANCEL</button>
          </Link>
        </fieldset>
      </form>
    </div>
  );
};

export default ShippingForm;
