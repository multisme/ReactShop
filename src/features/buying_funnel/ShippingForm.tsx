import React, { useState } from 'react'

export function useFormFields(initialState: any) {
  const [fields, setValues] = useState(initialState);

  return [
    fields,
    function(event: any) {
      setValues({
        ...fields,
        [event.target.id]: event.target.value
      });
    }
  ];
}

export const ShippingForm = () => {
        
        const [fields, handleFieldChange] = useFormFields({
               fistName: "",
               lastName: "",
               email: "",
               address: "",
               city: "",
               postalCode: ""
                });

        return (
        <div id="buy">
        <form>
                <label>First Name: <input type="text" value={fields.fistName} /> </label>
                <label>Last Name: <input type="text" value={fields.lastName} /> </label>
                <label>Email: <input type="email" value={fields.email} /> </label>
                <label>Address: <input type="text" value={fields.address} /> </label>
                <label>City: <input type="text" value={fields.city} /> </label>
                <label>Postal Code: <input type="text" value={fields.postalCode} /> </label>
        </form>
        </div>
       )
}
