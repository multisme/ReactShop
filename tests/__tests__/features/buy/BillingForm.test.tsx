import ShippingForm from "features/buy/ShippingForm"
import {shallow} from "enzyme"

describe("ShippingForm", () =>{
        it("renders", () => {
                const component = shallow(<ShippingForm />);
                expect(component).toMatchSnapshot()
        })
})
