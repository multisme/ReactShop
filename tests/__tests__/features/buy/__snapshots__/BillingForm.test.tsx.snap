// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingForm renders 1`] = `
<div
  className="rectangle"
  id="ShipForm"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name:
       
      <input
        id="name"
        onChange={[Function]}
        required={true}
        type="text"
      />
       
    </label>
    <label>
      Email:
       
      <input
        id="email"
        onChange={[Function]}
        required={true}
        type="email"
      />
       
    </label>
    <label>
      Shipping Address:
       
      <input
        id="address"
        onChange={[Function]}
        required={true}
        type="text"
      />
       
    </label>
    <label>
      City:
       
      <input
        id="city"
        onChange={[Function]}
        required={true}
        type="text"
      />
       
    </label>
    <label>
      Postal Code:
       
      <input
        id="postalCode"
        onChange={[Function]}
        required={true}
        type="text"
      />
       
    </label>
    <fieldset
      className="button"
    >
      <button
        onSubmit={[Function]}
        type="submit"
      >
        SHIP
      </button>
      <Link
        to="/home"
      >
        <button>
          CANCEL
        </button>
      </Link>
    </fieldset>
  </form>
</div>
`;
